src/api-endpoints.ts

```diff
@@ -10358,3 +10373,49 @@ export const search = {
   bodyParams: ["sort", "query", "start_cursor", "page_size", "filter"],
   path: (): string => `search`,
 } as const
+
+type CreateCommentBodyParameters =
+  | {
+      parent: { page_id: IdRequest; type?: "page_id" }
+      rich_text: Array<RichTextItemRequest>
+    }
+  | { discussion_id: IdRequest; rich_text: Array<RichTextItemRequest> }
+
+export type CreateCommentParameters = CreateCommentBodyParameters
+
+export type CreateCommentResponse =
+  | CommentObjectResponse
+  | PartialCommentObjectResponse
+
+export const createComment = {
+  method: "post",
+  pathParams: [],
+  queryParams: [],
+  bodyParams: ["parent", "rich_text", "discussion_id"],
+  path: (): string => `comments`,
+} as const
+
+type ListCommentsQueryParameters = {
+  block_id: IdRequest
+  start_cursor?: string
+  page_size?: number
+}
+
+export type ListCommentsParameters = ListCommentsQueryParameters
```

Comment:
@wcauchois why do we need to create an extra type just to set it equal to an existing type?

Reply:
this is just part of the codegen, it's all automatic. if we had body parameters as well as query parameters, this type would be `ListCommentsBodyParameters | ListCommentsQueryParameters` - so it makes more sense when it's a union like that.

the codegen isn't special cased to eliminate this alias if the endpoint only has one type of parameters. i don't really mind, it doesn't affect anything too negatively and keeps things consistent with other endpoints!